gui/qt/paytoedit.py

```diff
@@ -98,6 +98,20 @@ def parse_amount(self, x):
         p = pow(10, self.amount_edit.decimal_point())
         return int(p * PyDecimal(x.strip()))
 
+    def get_cash_account(self, username, block, collision):
```

Comment:
This function does not perform any of the verification steps that are required by the Cash Accounts spec, and merely trusts what the server returns.

* It must verify the SPV proof for the transaction.
* It must calculate the collision ID and verify that it matches the requested one.
* It must decode the registration transaction itself, and completely ignore the ['address'] field from the server's response. That decoded address response is a red herring that no wallet should ever use, and ought to be removed.
* The decoding process must implement all cash accounts rules to ensure it only accepts valid registrations:
  * Follows exactly the OP_RETURN number and OP_RETURN push forms decreed in specification.
  * The account name must follow the specified regex and length limits.

Reply:
Yeah agreed, verification is quite necessary, else the model is trusted.

Reply:
I'm not saying we should distrust Jonathan btw ... I'm saying we shouldn't trust the hacker who breaks into his system. :-D

Reply:
Very true. I believe Imaginary is working on a revised get_cash_account function right now, as I'm quite busy with homework at the moment, and I won't have much time in the coming days.

Reply:
the account/ lookup endpoint should never be used as the source of data, its purpose is to expose what the database have decoded as the data for the transaction to help implementers double-check and verify that their code is working.

Reply:
@monsterbitar I'd suggest removing that part of the API and instead make a 'test vectors' page. Or, randomly go in and change the insecure parts of API sometimes -- if wallets fail because of this change, good!

When I was writing my comment above I was rereading [the spec](https://gitlab.com/cash-accounts/specification/blob/master/SPECIFICATION.md) and noticed it doesn't have anywhere a concise and complete description of what wallets are expected to do. If you want people to implement cash accounts in their wallet then it would be a good idea to have those instructions -- use RFC language like "wallet MUST do X".